### PR TITLE
[Imporvement]: Add spotless:validate to core and hive module

### PR DESCRIPTION
**Pull request by baiyangtx**
## Why are the changes needed?

This is a subtask of #1402.  To replace `checkstyle` to `spotless` as default codestyle formatter.

## Brief change log

- Enable spotless configurations for `core` and `hive` modules.
- Skip `maven-checkstyle-plugin` for `core` and `hive` modules.
- Run `spotless:apply` for `core` and `hive` modules.

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (no)

### CODE REVIEW

File: hive/src/main/java/com/netease/arctic/hive/io/writer/AdaptHiveOutputFileFactory.java [hunk 1]
```diff
@@ -31,22 +31,11 @@
...
  *
- * For adapt hive table without partitions the dir construct is :
- *    ${table_location}
- *            -| change
- *            -| base
- *            -| hive
- *                  -| ${timestamp}_{txid}
- * txId of unkeyed table is random long.
+ * <p>For adapt hive table without partitions the dir construct is : ${table_location} -| change -|
+ * base -| hive -| ${timestamp}_{txid} txId of unkeyed table is random long.
```
Comment (zhongqishang) [inline hive/src/main/java/com/netease/arctic/hive/io/writer/AdaptHiveOutputFileFactory.java:38]: We should keep the original format.

File: hive/src/main/java/org/apache/iceberg/parquet/AdaptHiveParquetConversions.java [hunk 2]
```diff
@@ -85,43 +84,48 @@ static <T> Literal<T> fromParquetPrimitive(Type type, PrimitiveType parquetType,
...
-            .plusNanos(timeOfDayNanos);
+        Instant instant =
+            Instant.ofEpochMilli(TimeUnit.DAYS.toMillis(julianDay - 2_440_588L))
+                .plusNanos(timeOfDayNanos);
 
         if (!((Types.TimestampType) icebergType).shouldAdjustToUTC()) {
-          //iceberg org.apache.iceberg.expressions.Literals resolve timestamp without tz use UTC, but in fact it is
+          // iceberg org.apache.iceberg.expressions.Literals resolve timestamp without tz use UTC,
+          // but in fact it is
           // local time zone
```
Comment (zhongqishang) [inline hive/src/main/java/org/apache/iceberg/parquet/AdaptHiveParquetConversions.java:105]: In before this is via press Enter to format, but now it looks a bit strange after `spotless:apply`.

File: hive/src/main/java/com/netease/arctic/hive/io/writer/AdaptHiveOutputFileFactory.java (hunk 1 above)
Comment (baiyangtx) [reply hive/src/main/java/com/netease/arctic/hive/io/writer/AdaptHiveOutputFileFactory.java]: fixed.

File: hive/src/main/java/org/apache/iceberg/parquet/AdaptHiveParquetConversions.java (hunk 2 above)
Comment (baiyangtx) [reply hive/src/main/java/org/apache/iceberg/parquet/AdaptHiveParquetConversions.java]: fixed.

**Review by zhongqishang — APPROVED** [review]
LGTM.

**Review by wangtaohz — APPROVED** [review]
LGTM.

**Comment by shidayang** [comment]
The classes in org.* package of hive is copied from iceberg. should we spotless them?

**Pull request merged**